From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: qeth: pre z9 systems return HiperSocket version string different.
References: bnc#440610,LTC#49052

Symptom:     Device startup message shows invalid version string.
Problem:     z9 changed the string format from EBCDIC to ASCII.
Solution:    In case an EBCDIC HiperSocket version string is detected
             the string is converted to ASCII before insertion into
             message.

Acked-by: John Jolly <jjolly@suse.de>

Index: linux-sles11/drivers/s390/net/qeth_core_main.c
===================================================================
--- linux-sles11.orig/drivers/s390/net/qeth_core_main.c
+++ linux-sles11/drivers/s390/net/qeth_core_main.c
@@ -2239,7 +2239,8 @@ void qeth_print_status_message(struct qe
 		}
 		/* fallthrough */
 	case QETH_CARD_TYPE_IQD:
-		if (card->info.guestlan) {
+		if ((card->info.guestlan) ||
+		    (card->info.mcl_level[0] & 0x80)) {
 			card->info.mcl_level[0] = (char) _ebcasc[(__u8)
 				card->info.mcl_level[0]];
 			card->info.mcl_level[1] = (char) _ebcasc[(__u8)
